### PR TITLE
[CORE] Fix Spark-3.5 CI

### PR DESCRIPTION
**Pull request by ulysses-you**
## What changes were proposed in this pull request?

The current Spark3.5 gluten-it actually runs Spark3.2.2, this pr adds Spark3.5 profile for it. Besides, this pr fixes Spark3.5 compilation issue.

## How was this patch tested?

Pass CI

### CODE REVIEW

**Review by yaooqinn — APPROVED** [review]

**Review by JkSelf — APPROVED** [review]
LGTM. Thanks.

**Review by PHILO-HE — APPROVED** [review]
Thanks!

**Comment by ulysses-you** [comment]
the 3.5 CI has not yet passed...

File: tools/gluten-it/pom.xml [hunk 1]
```diff
@@ -61,12 +58,6 @@
         <artifactId>spark-catalyst_${scala.binary.version}</artifactId>
         <version>${spark.version}</version>
         <scope>provided</scope>
-        <exclusions>
-          <exclusion>
-            <groupId>org.apache.arrow</groupId>
-            <artifactId>*</artifactId>
```
Comment (ulysses-you) [inline tools/gluten-it/pom.xml:67]: gluten would shade arrow, so we do not need remove these for gluten-it any more

**Review by PHILO-HE — APPROVED** [review]

**Review by JkSelf — APPROVED** [review]

**Pull request merged**